backend/app/services/rag/pipeline/pandoc.py

```
@@ -0,0 +1,288 @@
+
+            # Read converted Markdown
+            markdown_content = Path(output_path).read_text(encoding="utf-8")
+            logger.info(
+                f"Pandoc conversion successful: {len(content)} bytes -> "
+                f"{len(markdown_content)} characters"
+            )
+
+            return markdown_content
+
+        except subprocess.TimeoutExpired:
+            raise PandocConversionError("Pandoc conversion timed out after 120 seconds")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)